sdv/single_table/base.py

```diff
@@ -383,7 +384,18 @@ def preprocess(self, data):
                 "please refit the model using 'fit' or 'fit_processed_data'."
             )
 
-        return self._preprocess(data)
+        for column in data.columns:
+            if isinstance(column, int):
+                self._original_columns = data.columns
+                data.columns = data.columns.astype(str)
+                break
+
+        preprocess_data = self._preprocess(data)
+
+        if not self._original_columns.empty:
+            data.columns = self._original_columns
```

Comment:
Can we split the logic of the `_original_columns` and the `preprocessing`? By having a private method like `_set_original_column_names(data)` or similar